src/garage/tf/algos/rl2.py

```diff
@@ -0,0 +1,316 @@
+
+        Returns:
+            dict: Processed sample data, with values:
+                numpy.ndarray: Observations. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Actions. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Rewards. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Terminal signals. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Returns. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Valids. Shape:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                numpy.ndarray: Lengths. Shape:
+                    :math:`[N, episode_per_task]`
+                dict[numpy.ndarray]: Environment Infos. Shape of values:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+                dict[numpy.ndarray]: Agent Infos. Shape of values:
+                    :math:`[N, episode_per_task * max_path_length, S^*]`
+
+        """
+        all_paths = []
+        concatenated_path_in_meta_batch = []
+        lengths = []
+        for _, path in paths.items():
+            concatenated_path, paths = self._concatenate_paths(path)
+            concatenated_path_in_meta_batch.append(concatenated_path)
+            all_paths.extend(paths)
+
+        (observations, actions, rewards, terminals, returns, valids, lengths,
+            env_infos, agent_infos) = \
+            self._stack_paths(
+                max_len=self._inner_algo.max_path_length,
+                paths=concatenated_path_in_meta_batch
+            )
+
+        (_observations, _actions, _rewards, _terminals, _returns, _valids,
+            _lengths, _env_infos, _agent_infos) = \
+            self._stack_paths(
+                max_len=self._max_path_length,
+                paths=all_paths
+            )
+
+        ent = np.sum(self._policy.distribution.entropy(agent_infos) *
+                     valids) / np.sum(valids)
+
+        # performance is evaluated per individual paths
+        undiscounted_returns = self.evaluate_performance(
+            itr,
+            dict(env_spec=self._env_spec,
+                 observations=_observations,
+                 actions=_actions,
+                 rewards=_rewards,
+                 terminals=_terminals,
+                 returns=_returns,
+                 env_infos=_env_infos,
+                 agent_infos=_agent_infos,
+                 lengths=_lengths,
+                 discount=self._discount,
+                 episode_reward_mean=self._inner_algo.episode_reward_mean))
+
+        self._inner_algo.episode_reward_mean.extend(undiscounted_returns)
+
+        # Log success rate for MetaWorld
```

Comment:
can you make an issue on metaworld to utilize termination conditions for this instead?

Reply:
also, can you add this to `RLAlgorithm.evaluate_performance` instead?

Reply:
What exactly does "utilize termination conditions for this" do?

Reply:
it means that `env_info['success']` and `done` should be the same thing.

Reply:
I think changing metaworld is probably the wrong approach given our timeline. At least, that's what I did for `log_performance` (which should obviate the need for this code).